File: src/jk_5/nailed/event/NailedEvent.java
package jk_5.nailed.event;

/**
 * No description given
 *
 * @author jk-5
 */
public class NailedEvent {

    private boolean canceled = false;

    public boolean isCancelable(){
        return false;
    }

    public void setCanceled(Boolean cancel){
        this.canceled = cancel;
    }

    public boolean isCanceled(){
        return this.isCancelable() && this.canceled;
    }
}

File: src/jk_5/nailed/event/NailedEvent.scala
package jk_5.nailed.event

/**
 * No description given
 *
 * @author jk-5
 */
class NailedEvent {
  private var canceled = false
  @inline def isCancelable = false
  @inline final def isCanceled = this.canceled
  @inline final def setCanceled(cancel: Boolean) = this.canceled = cancel
}
